File: Java/src/NextPermutationTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class NextPermutationTest {

    @Test
    public void testNextPermutation() {
        NextPermutation np = new NextPermutation();

        int[] array123 = new int[]{1, 2, 3};
        np.nextPermutation(array123);
        Assert.assertTrue(Arrays.equals(array123, new int[]{1, 3, 2}));

        int[] array321 = new int[]{3, 2, 1};
        np.nextPermutation(array321);
        Assert.assertTrue(Arrays.equals(array321, new int[]{1, 2, 3}));

        int[] array115 = new int[]{1, 1, 5};
        np.nextPermutation(array115);
        Assert.assertTrue(Arrays.equals(array115, new int[]{1, 5, 1}));
    }

}

File: Kotlin/src/NextPermutationTest.kt
import org.junit.Test

class NextPermutationTest {

    @Test
    fun testNextPermutation() {
        val np = NextPermutation()

        val array123 = intArrayOf(1, 2, 3)
        np.nextPermutation(array123)
        assert(array123.contentEquals(intArrayOf(1, 3, 2)))

        val array321 = intArrayOf(3, 2, 1)
        np.nextPermutation(array321)
        assert(array321.contentEquals(intArrayOf(1, 2, 3)))

        val array115 = intArrayOf(1, 1, 5)
        np.nextPermutation(array115)
        assert(array115.contentEquals(intArrayOf(1, 5, 1)))
    }

}